src/core/pubkey.zig

```diff
@@ -1,60 +1,50 @@
 const std = @import("std");
 const sig = @import("../sig.zig");
-const Ed25519 = std.crypto.sign.Ed25519;
-const U8ArrayConfig = sig.bincode.int.U8ArrayConfig;
 
 pub const Pubkey = extern struct {
     data: [size]u8,
+    const Self = @This();
 
     pub const size = 32;
-    pub const @"!bincode-config:data" = U8ArrayConfig(size);
+    pub const @"!bincode-config:data" = sig.bincode.int.U8ArrayConfig(size);
+
+    pub const zeroes: Pubkey = .{ .data = .{0} ** size };
```

Comment:
the reason we use default is because in agave the rust client derives the default value, which is all zeros, and has logic for that - all_zeros almost says intentionally, you want all zeros pubkey - whereas we actually dont care about the value, we just want the default, hence the current naming

Reply:
I don't think it makes sense to use the term "default" except in very specific circumstances (ie configuration of some algorithm with certain generally desirable characteristics whose details don't entirely matter, ie `std.Random.DefaultPrng`).

What does it mean for a Pubkey to be "default"? I could think of numerous ways in which to interpret such a value, zero-init being only one of them - it could just as well be some specifically designated value which the solana protocol has deemed "default", or any other kind of trivial init, like all-ones.

What benefit is there to using a *less* specific name? When the programmer reads `= Pubkey.zeroes`, they don't have to guess about anything, and can't make any wrong assumptions; whereas with `Pubkey.default`.

Just to cement the point, I would point out that in our own code, we actually *do* care about the value of a "defaulted" pubkey, this had a method called "isDefault" - I've changed it to "isAllZeroes", which better describes what we're actually checking for.

Reply:
thats a great point - thanks for explaining

Reply:
4d57500b7adc0d5c4166ffe7f70d8a2f560b56a8